pkgs/development/compilers/chicken/5/overrides.nix

```diff
@@ -68,13 +66,11 @@ in {
   soil = addToPropagatedBuildInputsWithPkgConfig pkgs.libepoxy;
   sqlite3 = addToBuildInputs pkgs.sqlite;
   stemmer = old:
-    # Undefined symbols for architecture arm64: "_sb_stemmer_delete"
-    (brokenOnDarwin old)
-    // (addToBuildInputs pkgs.libstemmer old);
+    (addToBuildInputs pkgs.libstemmer old)
+    // (addToCscOptions "-L -lstemmer" old);
   stfl = old:
-    # Undefined symbols for architecture arm64: "_clearok"
-    (brokenOnDarwin old)
-    // (addToBuildInputs [ pkgs.ncurses pkgs.stfl ] old);
+    (addToBuildInputs [ pkgs.ncurses pkgs.stfl ] old)
+    // (addToCscOptions "-L -lncurses" old);
```

Comment:
Why `-L`

```
    {
       # TODO: Weird the ld wrapper didn't find the libs
       postPatch = old.postPatch or "" + "substituteInPlace build-stfl.scm --replace 'pkg-config --libs stfl' 'pkg-config --libs stfl ncurses'";
     }
     // (addToBuildInputsWithPkgConfig [ pkgs.ncurses pkgs.stfl ] old);
```

https://github.com/Et7f3/nixpkgs/commit/3564e9f77485a60a69a5b71d2bc1dac0314fbcba

Reply:
`-L OPTION` is the way to pass options to linkers from `csc`. But yes, your solution is the best.